### PR TITLE
Add `DocumentationBuilder::with_standard_argument` to reduce copy/paste

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Part of https://github.com/apache/datafusion/issues/12740

## Rationale for this change

@Omega359 has created the basic framework for automatically generating content in #12668

I would like to get the code into shape where we can start cranking to port all the docs over and to do that we should get a good pattern in place

There is a lot of copy/paste generic wording for arguments, so let's make it easier to add details in the future

## What changes are included in this PR?
1. Add `DocumentationBuilder::with_standard_argument` to reduce copy/paste
2. Update code to use the new api

## Are these changes tested?
Yes, by CI

### CODE REVIEW

File: datafusion/functions-aggregate/src/bit_and_or_xor.rs [hunk 1]
```diff
@@ -142,10 +142,7 @@ fn get_bit_and_doc() -> &'static Documentation {
             .with_doc_section(DOC_SECTION_GENERAL)
             .with_description("Computes the bitwise AND of all non-null input values.")
             .with_syntax_example("bit_and(expression)")
-            .with_argument(
-                "expression",
-                "Expression to operate on. Can be a constant, column, or function, and any combination of arithmetic operators.",
-            )
+            .with_standard_argument("expression", "Integer")
```
Comment (alamb) [inline datafusion/functions-aggregate/src/bit_and_or_xor.rs:145]: The idea is to reduce this the boiler plate here

File: docs/source/user-guide/sql/aggregate_functions_new.md [hunk 2]
```diff
@@ -47,7 +47,7 @@ bit_and(expression)
 
 #### Arguments
 
-- **expression**: Expression to operate on. Can be a constant, column, or function, and any combination of arithmetic operators.
+- **expression**: Integer expression to operate on. Can be a constant, column, or function, and any combination of operators.
```
Comment (alamb) [inline docs/source/user-guide/sql/aggregate_functions_new.md:50]: I think the arithmetic is superflous in this context

File: datafusion/functions/src/regex/regexplike.rs [hunk 3]
```diff
@@ -67,10 +67,9 @@ SELECT regexp_like('aBc', '(b|d)', 'i');
 ```
 Additional examples can be found [here](https://github.com/apache/datafusion/blob/main/datafusion-examples/examples/regexp.rs)
 "#)
-            .with_argument("str",
-                           "String expression to operate on. Can be a constant, column, or function, and any combination of string operators.")
-            .with_argument("regexp",
-                           "Regular expression to test against the string expression. Can be a constant, column, or function.")
+            .with_standard_argument("str",
+                           "String")
+            .with_argument("regexp","Regular")
```
Comment (Omega359) [inline datafusion/functions/src/regex/regexplike.rs:72]: with_standard_argument

Comment (alamb) [reply datafusion/functions/src/regex/regexplike.rs]: Fixed in 3e816e14c

File: datafusion/expr/src/udf_docs.rs [hunk 4]
```diff
@@ -142,6 +143,23 @@ impl DocumentationBuilder {
...
 
+    /// Add a standard "expression" argument to the documentation
+    ///
+    /// This is a common argument for scalar UDFs that operate on an expression and is rendered like
+    ///
+    /// ```text
+    /// <arg_name>:
+    ///   <expression_type> expression to operate on. Can be a constant, column, or function, and any combination of arithmetic operators.
+    /// ```
+    pub fn with_standard_argument(
```
Comment (comphead) [inline datafusion/expr/src/udf_docs.rs:154]: it might be useful to mention how standard argument differs from just argument?

**Review by comphead — APPROVED** [review]
love it, gradually it comes similar to what Spark have reg to examples/help/docs 💪

File: datafusion/expr/src/udf_docs.rs (hunk 4 above)
Comment (alamb) [reply datafusion/expr/src/udf_docs.rs:160]: Good idea, done in 09a4b7bb0

**Comment by alamb** [comment]
Thanks again @Omega359 and @comphead

**Pull request merged**